### DISCUSSION TITLE
Suggestion: new windows should open in the same desktop that currently has focus

### DISCUSSION THREAD

**Discussion by gongfarmer** [asmvik/yabai#1525] [Ideas]
I am using a multi-monitor setup with yabai and multiple desktops.

When I open an app by clicking the dock or using spotlight, it is apparently random which monitor the new application window will open up on.   This is a behaviour of stock MacOS, not of yabai.   Is it possible for yabai to improve this behaviour?

I am able to configure rules to specify that a particular application must open on a particular workspace using[ this procedure](https://www.lifewire.com/assign-mac-apps-to-open-in-specific-desktop-space-2260812) which is built in to Mac OS.

However my suggestion is about changing the default behaviour, which seems surprising.  Application windows should open in the same desktop that currently has focus (ie. where the mouse is, usually.)

**Comment by asmvik** [comment]
https://github.com/koekeishiya/yabai/blob/master/doc/yabai.asciidoc

>window_origin_display [default|focused|cursor]
 Specify which display a newly created window should be managed in.
 default: The display in which the window is created (standard macOS behaviour).
 focused: The display that has focus when the window is created.
 cursor: The display that currently holds the mouse cursor.
